app/components/network_indicator/network_indicator.js

```diff
@@ -124,21 +125,25 @@ export default class NetworkIndicator extends PureComponent {
     }
 
     connect = (displayBar = false) => {
+        const {connection} = this.props.actions;
         clearTimeout(this.connectionRetryTimeout);
 
         NetInfo.isConnected.fetch().then(async (isConnected) => {
             const {hasInternet, serverReachable} = await checkConnection(isConnected);
 
-            this.props.actions.connection(hasInternet);
+            connection(hasInternet);
+            this.hasInternet = hasInternet;
```

Comment:
@enahum I'm curious why instance fields (instead of static or state) are allowed here. Does it work as expected in React component lifecycle?

Reply:
yes it does.. this are just there to control the event handler for network detection as for some reason RN seems to be emitting that event twice